activerecord/test/cases/connection_adapters/connection_handler_test.rb

```diff
@@ -71,6 +71,54 @@ def test_establish_connection_using_3_levels_config
         ENV["RAILS_ENV"] = previous_env
       end
 
+      unless in_memory_db?
+        def test_establish_connection_using_3_level_config_defaults_to_default_env_primary_db
+          previous_env, ENV["RAILS_ENV"] = ENV["RAILS_ENV"], "default_env"
+
+          config = {
+            "default_env" => {
+              "primary"  => { "adapter" => "sqlite3", "database" => "db/primary.sqlite3" },
+              "readonly" => { "adapter" => "sqlite3", "database" => "db/readonly.sqlite3" }
+            },
+            "another_env" => {
+              "primary"  => { "adapter" => "sqlite3", "database" => "db/another-primary.sqlite3" },
+              "readonly" => { "adapter" => "sqlite3", "database" => "db/another-readonly.sqlite3" }
+            }
+          }
+          @prev_configs, ActiveRecord::Base.configurations = ActiveRecord::Base.configurations, config
+
+          ActiveRecord::Base.establish_connection
+
+          assert_equal "db/primary.sqlite3", ActiveRecord::Base.connection.pool.spec.config[:database]
```

Comment:
`ActiveRecord::Base.connection.pool.spec.config[:database]` creates extra `rails/activerecord/db/primary.sqlite3` and  this test fails if run via `bin/test`, the same problem with the test that below in this file.

Reply:
What the what. This works fine with `bundle exec ruby -Ilib:test test/cases/connection_adapters/connection_handler_test.rb` 🤔

Reply:
Anyway there's another bug in here that I need to look at and will fix on Monday.